### PR TITLE
Patterned mesh

### PR DESCRIPTION
**Pull request by jwpeterson**
Replaces #6831.

Adds Doxygen comments, class description, and whitespace fixes as requested by reviewers.

### CODE REVIEW

**Comment by jwpeterson** [comment]
Looks like we ran out of space on the build box...

```
/tmp/ccL54vMf.s: Assembler messages:
/tmp/ccL54vMf.s: Fatal error: can't write richards/src/kernels/.libs/RichardsFlux.x86_64-unknown-linux-gnu.dbg.o: No space left on device
/tmp/ccL54vMf.s: Fatal error: can't close richards/src/kernels/.libs/RichardsFlux.x86_64-unknown-linux-gnu.dbg.o: No space left on device
```

File: framework/src/mesh/PatternedMesh.C [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+template<>
+InputParameters validParams<PatternedMesh>()
+{
+  InputParameters params = validParams<MooseMesh>();
+  params.addRequiredParam<std::vector<MeshFileName> >("files", "The name of the mesh files to read.  They are automatically assigned ids starting with zero.");
+
+  params.addParam<Real>("x_width", 0, "The tile width in the x direction");
+  params.addParam<Real>("y_width", 0, "The tile width in the y direction");
+  params.addParam<Real>("z_width", 0, "The tile width in the z direction");
```
Comment (permcody) [inline framework/src/mesh/PatternedMesh.C]: We really should use addCheckedRangeParam here to make sure these are positive reals.

Comment (jwpeterson) [inline framework/src/mesh/PatternedMesh.C]: OK, I can do this real quick.

**Comment by friedmud** [comment]
Thanks @jwpeterson !  I was going to get tooth of these this week... but it lookalike you guys got tired of waiting :smile:

**Comment by permcody** [comment]
👍

**Comment by permcody** [comment]
Ah @friedmud - it's good to know nothing has changed. One commit to the framework and we aren't valgrind clean....
https://www.moosebuild.org/job/22221/

**Comment by jwpeterson** [comment]
This is the bug, loop needs to go to `_row_meshes.size()`
```
  for (unsigned int i = 1; i < _meshes.size(); i++)
    delete _row_meshes[i];
```

**Comment by jwpeterson** [comment]
I will put in a PR to fix that

**Comment by permcody** [comment]
Let's shrink and regold this monster too. This test is too big and timing out in parallel debug mode.

**Comment by jwpeterson** [comment]
> Let's shrink and regold this monster too.

OK, I'll do that on #6964 too.

**Pull request merged**